hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -193,7 +193,7 @@ String & String::copy(const __FlashStringHelper *pstr, unsigned int length)
 void String::move(String &rhs)
 {
 	if (buffer) {
-		if (capacity >= rhs.len) {
+		if (rhs && capacity >= rhs.len) {
 			strcpy(buffer, rhs.buffer);
 			len = rhs.len;
 			rhs.len = 0;
```

Comment:
@matthijskooijman good question!

Maybe it's to avoid memory fragmentation?

@PaulStoffregen @damellis @cmaglie do you have any thoughts on this?